### PR TITLE
Updates to upgrade tests

### PR DESCRIPTION
**Pull request by t0yv0**
Continuing to evolve providertest library; this is introducing the preferred pattern from that library, breaking out top-level tests by test-program and combining upgrades with other forms of testing (e2e, sdk).

Curious about CI results here; this may need a little more support from the Makefile and CI system since these tests run in the `prerequisites` phase which may not have the necessary credentials available to test `pulumi preview`.

### CODE REVIEW

**Comment by t0yv0** [comment]
Blocked by https://github.com/pulumi/ci-mgmt/issues/675

**Comment by t0yv0** [comment]
This is now upgraded to 0.0.2 of the provider and passing locally for me. I think there still is a problem with running these in CI. Perhaps I can find a way to skip them in CI until we can circle back to reintroducing creds into the prerequisites job.

**Comment by t0yv0** [comment]
I've worked around the issues with lack of credentials. We can check this in, I think it will help with having the upgrade tests aligned with the latest shape of providertest. CC @guineveresaenger

File: provider/provider_test.go
Comment (mjeffryes) [inline provider/provider_test.go:132]: I thought that one of the features of the providertest API was supposed to be that we can easily choose which kinds of tests to run? Can we not just set some flags to only run upgrade tests?

**Review by mjeffryes — APPROVED** [review]
Seems like a reasonable step to unblock the state of testing in GCP

File: provider/provider_test.go
Comment (t0yv0) [reply provider/provider_test.go:132]: Yeah, we missed making this flag settable at Go level.

**Review by guineveresaenger — APPROVED** [review]
Happy to try this.

Thank you for these changes!

**Pull request merged**
